src/modules/70-pipeline/components/Volumes/Volumes.tsx

```diff
@@ -0,0 +1,372 @@
+  HostPath = 'HostPath',
+  PersistentVolumeClaim = 'PersistentVolumeClaim'
+}
+const getTypeOptions = (getString: UseStringsReturn['getString']): { label: string; value: VolumesTypes }[] => [
+  { label: getString('pipeline.buildInfra.emptyDirectory'), value: VolumesTypes.EmptyDir },
+  { label: getString('pipeline.buildInfra.hostPath'), value: VolumesTypes.HostPath },
+  { label: getString('pipeline.buildInfra.persistentVolumeClaim'), value: VolumesTypes.PersistentVolumeClaim }
+]
+const getDefaultSpec = (type: VolumesTypes): { [key: string]: any } => {
+  if (type === VolumesTypes.EmptyDir) {
+    return { medium: '' }
+  } else if (type === VolumesTypes.HostPath) {
+    return { path: '' }
+  } else if (type === VolumesTypes.PersistentVolumeClaim) {
+    return { claimName: '', readOnly: false }
+  }
+  return {}
+}
+
+const getAdditionalTextFields = (
+  type: VolumesTypes,
+  getString: UseStringsReturn['getString']
+): { label: string; name: string; tooltipId: string; optional: boolean; fieldType: FieldType }[] | undefined => {
+  if (type === VolumesTypes.EmptyDir) {
+    return [
+      {
+        label: getString('connectors.cdng.verificationSensitivityLabel.medium'),
+        name: 'spec.medium',
+        fieldType: FieldType.TextInput,
+        tooltipId: 'medium',
+        optional: true
+      },
+      {
+        label: getString('pipeline.buildInfra.size'),
+        name: 'spec.size',
+        fieldType: FieldType.TextInput,
+        tooltipId: 'size',
+        optional: true
+      }
+    ]
+  } else if (type === VolumesTypes.HostPath) {
+    return [
+      {
+        label: getString('common.path'),
+        name: 'spec.path',
+        fieldType: FieldType.TextInput,
+        tooltipId: 'path',
+        optional: false
+      },
+      {
+        label: getString('pipeline.buildInfra.pathType'),
+        name: 'spec.type',
+        fieldType: FieldType.TextInput,
+        tooltipId: 'pathType',
+        optional: true
+      }
+    ]
+  } else if (type === VolumesTypes.PersistentVolumeClaim) {
+    return [
+      {
+        label: getString('pipeline.buildInfra.claimName'),
+        name: 'spec.claimName',
+        fieldType: FieldType.TextInput,
+
+        tooltipId: 'claimName',
+        optional: false
+      },
+      {
+        label: getString('common.readOnly'),
+        name: 'spec.readOnly',
+        fieldType: FieldType.Checkbox,
+        tooltipId: 'readOnly',
+        optional: false
+      }
+    ]
+  }
+}
+
+export const getOptionalSubLabel = (getString: UseStringsReturn['getString'], tooltip?: string) => (
+  <Text
+    tooltipProps={tooltip ? { dataTooltipId: tooltip } : {}}
+    color={Color.GREY_400}
+    font={{ size: 'small', weight: 'semi-bold' }}
+    style={{ textTransform: 'capitalize' }}
+  >
+    {getString?.('common.optionalLabel')}
+  </Text>
+)
+
+const renderFields = ({
+  formik,
+  getString,
+  type,
+  index,
+  expressions,
+  allowableTypes,
+  disabled
+}: {
+  formik: any
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232285)
```suggestion
  formik: unknown
```